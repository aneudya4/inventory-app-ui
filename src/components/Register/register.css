.validation-errors {
  color: tomato;
}
label span {
  color: tomato;
}

input {
  outline: none;
}
