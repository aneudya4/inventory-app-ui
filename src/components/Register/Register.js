import React, { useCallback, useContext, useState } from 'react';
import apiContext from '../../apiContext';
import config from '../config';
import { Redirect } from 'react-router-dom';

import app from '../../firebaseConfig';
import { Link } from 'react-router-dom';
import './register.css';
const Register = ({ history }) => {
  const { user } = useContext(apiContext);
  const [validateForm, setValidateForm] = useState(false);
  const handleRegister = useCallback(
    async (e) => {
      e.preventDefault();
      const { name, email, password } = e.target.elements;
      console.log(name.value, email.value, 'HERE MMGF');
      if (password.value.length < 6) {
        setValidateForm(true);
        return;
      }
      try {
        await app
          .auth()
          .createUserWithEmailAndPassword(email.value, password.value)
          .then(function (result) {
            return result.user.updateProfile({
              displayName: name.value,
            });
          });
        await postUser(name.value, email.value);
        history.push('/accounts/login');
      } catch (error) {
        alert('error');
      }
    },
    [history]
  );
  const postUser = async (name, email) => {
    await fetch(`${config.API_ENDPOINT}/users`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        Authorization: `Bearer ${config.API_KEY}`,
      },
      body: JSON.stringify({ name, email }),
    });
  };

  if (user) {
    return <Redirect to='/auth/dashboard/overview' />;
  }
  return (
    <div className='register-form form'>
      <form onSubmit={handleRegister}>
        <h3>Register</h3>
        {validateForm && (
          <span className='validation-errors'>
            *Password has to be at least 6 characters
          </span>
        )}
        <label htmlFor='email'>
          Your Name<span>*</span>
          <input
            id='name'
            name='name'
            type='text'
            placeholder='John Doe'
            required
            maxLength='20'
          />
        </label>
        <label htmlFor='email'>
          Your Email<span>*</span>
          <input
            id='email'
            name='email'
            type='email'
            placeholder='JohnDoe@email.com'
            required
          />
        </label>
        <label htmlFor='password'>
          Your Passwords<span>*</span>
          <input
            id='password'
            name='password'
            type='password'
            placeholder='Create a Password'
            required
          />
        </label>
        <button className='btn'>Register</button>
        <span>
          Already have an account? <Link to='./login'>Log in</Link>
        </span>
      </form>
    </div>
  );
};

export default Register;
