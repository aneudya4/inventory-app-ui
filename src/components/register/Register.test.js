import { shallow } from 'enzyme';

import Register from './Register.js';

describe('Test on the <Register/> Component', () => {
  it('renders Register without crashing', () => {
    const wrapper = shallow(<Register />);
    const RegisterButton = wrapper.find('button').at(0);
    expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
    expect(RegisterButton.text()).toBe('Register');
  });
});
