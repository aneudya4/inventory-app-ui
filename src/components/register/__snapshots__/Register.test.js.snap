// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <Register/> Component renders Register without crashing 1`] = `
<div
  className="register-form form"
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Register
    </h3>
    <label
      htmlFor="email"
    >
      Your Name
      <span>
        *
      </span>
      <input
        autoComplete="off"
        id="name"
        maxLength="20"
        name="name"
        placeholder="John Doe"
        required={true}
        type="text"
      />
    </label>
    <label
      htmlFor="email"
    >
      Your Email
      <span>
        *
      </span>
      <input
        autoComplete="off"
        id="email"
        name="email"
        placeholder="JohnDoe@email.com"
        required={true}
        type="email"
      />
    </label>
    <label
      htmlFor="password"
    >
      Your Passwords
      <span>
        *
      </span>
      <input
        autoComplete="off"
        id="password"
        name="password"
        placeholder="Create a Password"
        required={true}
        type="password"
      />
    </label>
    <button
      className="btn"
    >
      Register
    </button>
    <span>
      Already have an account? 
      <Link
        to="./login"
      >
        Log in
      </Link>
    </span>
  </form>
</div>
`;
